src/formats/XTC.cpp

```diff
@@ -0,0 +1,88 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include "chemfiles/formats/XTC.hpp"
+#include "chemfiles/files/XTCFile.hpp"
+
+#include "chemfiles/ErrorFmt.hpp"
+#include "chemfiles/Frame.hpp"
+using namespace chemfiles;
+
+template <> FormatInfo chemfiles::format_information<XTCFormat>() {
+    return FormatInfo("XTC").with_extension(".xtc").description("XTC binary format");
+}
+
+#define STRING_0(x) #x
+#define STRING(x) STRING_0(x)
+#define CHECK(x) check_xtc_error((x), (STRING(x)))
+
+XTCFormat::XTCFormat(std::string path, File::Mode mode, File::Compression compression)
+    : xtc_(std::move(path), mode) {
+    if (compression != File::DEFAULT) {
+        throw format_error("XTC format does not support compression");
+    }
+}
+
+size_t XTCFormat::nsteps() { return static_cast<size_t>(xtc_.get_nframes()); }
+
+void XTCFormat::read_step(size_t step, Frame& frame) {
+    step_ = static_cast<int64_t>(step);
+    CHECK(xdr_seek(xtc_, xtc_.get_offset(step_), SEEK_SET));
+    read(frame);
+}
+
+void XTCFormat::read(Frame& frame) {
+    int natoms = xtc_.get_natoms();
+    int md_step = 0;
+    float time = 0;
+    matrix box;
+    rvec* x = new rvec[natoms];
```

Comment:
This can be replaced by using a std::vector and using [std::vector::data](https://en.cppreference.com/w/cpp/container/vector/data) when calling the C function. This removes the need for manual memory management (the current version of the code is using free on a `new` allocated object, which is UB)

Reply:
This doesn't seem to work. I guess that's because the 3D array doesn't have a default value and this breaks stuff. Added proper delete for now.

Reply:
Isn't rvec just a typedef to `float[3]`? If so, you can create a std::vector<float> of size 3*natoms, and use that.

Reply:
Works now, must have messed something up yesterday.

Reply:
> It must be vector of type float[3].

You can use `reinterpret_cast<float[3]*>(float*)`

Reply:
Nice little trickery, didn't know about that =)